doc/timelock/deposit.tla

```diff
@@ -0,0 +1,102 @@
+(*
+ * Deposit method monitors for the Timelock contract
+ *
+ * Andrey Kuprianov, Jure Kukovec, 2024
+ *)
+---- MODULE deposit ----
+
+EXTENDS state, typedefs
+
+(*
+    @typeAlias: depositArgs = { 
+        from: $address,
+        token: $address,
+        amount: Int,
+        claimants: Seq($address),
+        time_bound: $timeBound
+    };
+*)
+deposit_typedefs == TRUE
+
+\* @type: ($depositArgs) => Bool;
+MustFail_deposit_TooManyClaimants(args) == 
+    Len(args.claimants) > 10
+
+\* We don't support signatures ATM, see #85
+\* $type: ($depositArgs, $env) => Bool;
+\* MustFail_deposit_Unauthorized(args, env) == 
+    \* ~authorized(args.from, env)
+
+\* Balance is externally relevant; Init flag is not
+\* What matters for users is that balance is not overwritten
+\* @type: ($env) => Bool;
+MustFail_deposit_AlreadyInitialized(env) == 
+    instance_has("Balance", env)
+
+\* @type: ($env) => Bool;
+MustHold_deposit_BalanceRecordCreated(env) ==
+    next_instance_has("Balance", env)
+
+\* @type: ($depositArgs) => Bool;
+MustHold_deposit_BalanceRecordCorrect(args) ==
+    /\ Balance'.token = args.token
+    /\ Balance'.amount = args.amount
+    /\ Balance'.time_bound = args.time_bound
+    /\ Balance'.claimants = args.claimants
+
+\* Everything below is deterministic, and will be generated automatically
+\* For now, we encode this manually
```

Comment:
Can we make this more visible?

The point is that monitors are small, so we should clearly separate the manual instrumentation part.